packages/dds/tree/.eslintrc.cjs

```diff
@@ -39,4 +39,33 @@ module.exports = {
 			},
 		},
 	],
+	settings: {
```

Comment:
@Josmithr @tylerbutler as resident eslint experts :). By adding these settings here, I'm seeing the behavior for `import/no-deprecated` change, and it seems like the baseline is wrong. One thing that seems weird to me is that the `minimal-deprecated.js` config extends `base.js`, but both have defined resolvers with minimal-deprecated seeming less appropriate:

```js
	settings: {
		"import/extensions": [".ts", ".tsx", ".d.ts", ".js", ".jsx"],
		"import/parsers": {
			"@typescript-eslint/parser": [".ts", ".tsx", ".d.ts"],
		},
		"import/resolver": {
			node: {
				extensions: [".ts", ".tsx", ".d.ts", ".js", ".jsx"],
			},
		},
		"jsdoc": {
			// The following are intended to keep js/jsx JSDoc comments in line with TSDoc syntax used in ts/tsx code.
			tagNamePreference: {
				arg: {
					message: "Please use @param instead of @arg.",
					replacement: "param",
				},
				argument: {
					message: "Please use @param instead of @argument.",
					replacement: "param",
				},
				return: {
					message: "Please use @returns instead of @return.",
					replacement: "returns",
				},
			},
		},
	},
```

vs the one in base:

```js
	settings: {
		"import/extensions": [".ts", ".tsx", ".d.ts", ".js", ".jsx"],
		"import/parsers": {
			"@typescript-eslint/parser": [".ts", ".tsx", ".d.ts"],
		},
		"import/resolver": {
			typescript: {
				extensions: [".ts", ".tsx", ".d.ts", ".js", ".jsx"],
			},
		},
	},
```

is there a reason for this? Any idea why we'd have strange behavior here? Seems concerning to me.

Reply:
oh, I see it looks like we write resolved configurations to common/build/eslint-config-fluid/printed-configs, and it ends up with an entry for node as well as typescript. So maybe the change here is adjusting which resolver is used for a given import, which affects which symbols are marked as deprecated?

Reply:
How have you been testing the changes? If you've been doing somehting like symlinking the shared eslint config to the client release group, then I have found that the import/ eslint rules behave very strangely. I've had some luck using pnpm overrides and a file: dependency on the shared config.

If you're seeing this behavior with local eslint config overrides, then I'm very confused why that's happening.

Reply:
Sorry, to be more clear: it seems like `import/no-deprecated` isn't working correctly on the sequence package when run in main. Notably, there are a number of deprecated merge-tree exports which sequence uses but has no corresponding eslint-disable for. This is the top of sequence.ts on main:

![image](https://github.com/microsoft/FluidFramework/assets/5222607/df47c1a9-518f-4c59-a91b-0c532e290c60)

E.g. SegmentGroup is one such export.

When I added this setting to sequence, it seems to have somehow fixed the `import/no-deprecated` such that it now correctly complains on usage of deprecated types, hence the changes in this PR to the sequence package.

Reply:
I think my plan here is:

1. Revert changes to base eslint config (as per other comment) in this PR. That should put it in a check-in-able state where our packages better represent their public API.
2. Investigate why import/no-deprecated seems like it's broken with current lint configuration
3. Make change to base eslint config in a separate PR, perhaps incorporating results of the above investigation
4. Integrate new eslint-fluid version in other packages
5. Remove overrides in sequence/tree that are rendered unnecessary by (4)

Reply:
I filed [AB#7812](https://dev.azure.com/fluidframework/internal/_workitems/edit/7812) to track (2).